File: LabPoo/src/main/encapsulamiento.ts
import { Humano, IUsuario } from "./interfaces";

class Persona implements Humano {
    public nombre: string;
    public edad: number;
    public sexo: string;        

    constructor(nombre: string, edad: number, sexo: string) {
        this.nombre = nombre;
        this.edad = edad;
        this.sexo = sexo;
    }
    public language(): void {
        console.log('Todos somos iguales c:')
    }

    public getNombre(): string {
        return this.nombre;
    }

    public getEdad(): number {
        return this.edad;
    }
    
    public getSexo(): string {
        return this.sexo;
    }

    public setNombre(nombre: string): void {
        this.nombre = nombre;
    }

    public setEdad(edad: number): void {
        this.edad = edad;
    }

    public setSexo(sexo: string): void {
        this.sexo = sexo;
    }    
}

abstract class Usuario implements IUsuario {
    constructor(
        public nombre: string, 
        public edad: number, 
        public sexo: string, 
        public rol: string
    ) {}
  
    abstract describir(): string;
  
    static saludar(): string {
      return 'Bienvenido al sistema de usuarios';
    }
  }

export { Persona, Usuario };

File: tests/src/encapsulamiento.py
# encapsulamiento.py
class Persona:
    def __init__(self, nombre: str, edad: int):
        self.__nombre = nombre
        self.__edad = edad

    def get_nombre(self) -> str:
        return self.__nombre

    def get_edad(self) -> int:
        return self.__edad

    def set_nombre(self, nombre: str) -> None:
        self.__nombre = nombre

    def set_edad(self, edad: int) -> None:
        self.__edad = edad

    def presentar(self) -> str:
        return f"Soy {self.get_nombre()} y tengo {self.get_edad()} años."
    
class Planta:
    def __init__(self, nombre: str, origen: str, clima: str):
        self.__nombre = nombre
        self.__origen = origen
        self.__clima = clima

    def get_nombre(self) -> str:
        return self.__nombre

    def get_origen(self) -> str:
        return self.__origen

    def get_clima(self) -> str:
        return self.__clima
    